tbmalt/physics/dftb/feeds.py

```diff
@@ -562,82 +569,74 @@ class SkFeed(IntegralFeed):
         instances will identify and set all NaNs to zero.
 
     """
-
-    # TODO:
-    #   - Split into VCR and non-VCR versions.
-    #   - Ensure interpolation agnosticism.
-    #   - Fully integrate with PyTorch "Module" system.
-
-    def __init__(self, on_sites: Dict[int, Union[Parameter, Tensor]],
-                 off_sites: Dict[Tuple[int, int, int, int], Union[CubicSpline, PolyInterpU]],
+    def __init__(self, on_sites: ParameterDict[str, Parameter],
+                 off_sites: ModuleDict[str, Module],
                  dtype: Optional[torch.dtype] = None,
                  device: Optional[torch.device] = None):
 
-        # Ensure that the on_sites are torch tensors
-        if not isinstance(temp := list(on_sites.values())[0], Tensor):
-            on_sites = {k: torch.tensor(v, dtype=dtype, device=device)
-                        for k, v in on_sites.items()}
+        temp = next(iter(on_sites.values()))
+        super().__init__(temp.dtype if dtype is None else dtype,
+                         temp.device if device is None else device)
```

Comment:
Hey Adam, I am wondering why these two lines are repeated again at line 597-598. Is there any special purpose? Thanks.

Reply:
That is an oversight on my part. I will remove the duplicate on lines 597-598.

Reply:
Nice thanks Adam:) BTW, it's the same for the VcrSkFeed.